File: Python/find_second_largest.py
def find_second_largest(arr, n):
    if n < 2:
        return -1
    first = second = float('-inf')
    for i in range(n):
        if arr[i] > first:
            second = first
            first = arr[i]
        elif arr[i] > second and arr[i] != first:
            second = arr[i]
    return second

if __name__ == "__main__":
    arr = [12, 35, 1, 10, 34, 1]
    n = len(arr)
    print(f"Second largest element: {find_second_largest(arr, n)}")

File: C/find_second_largest.c
#include <stdio.h>

int findSecondLargest(int arr[], int n) {
    int first = arr[0], second = -1;
    for(int i = 1; i < n; i++) {
        if(arr[i] > first) {
            second = first;
            first = arr[i];
        }
        else if(arr[i] > second && arr[i] != first) {
            second = arr[i];
        }
    }
    return second;
}

int main() {
    int arr[] = {12, 35, 1, 10, 34, 1};
    int n = sizeof(arr)/sizeof(arr[0]);
    int result = findSecondLargest(arr, n);
    printf("Second largest element: %d\n", result);
    return 0;
}